### DISCUSSION TITLE
FinLine vs FinPolyLine visibility after refresh

### DISCUSSION THREAD

**Discussion by seyeint** [highfestiva/finplot#449] [Q&A] [answered]
This is a print scrn of my program where we can see 3 lines but only the FinLine stays after I advance one candle in time.
[First one](https://prnt.sc/_naloM2S13jl)
and 
[Second one](https://prnt.sc/MiO9bTVSstSm)
Can you please enlighten me on 1) how to change visibility for the one that stays and more important 2) keep visibility for FinPolyLine that goes away?

**Comment by highfestiva** [comment]
No, but possibly with a minimum of code I might see what's going on and why.

**Comment by seyeint** [comment]
i create the interactive finpolyline like this:
_ln_ = fplt.add_line((date, price), (relative_to_15m[combo.currentText()] * (10 ** 9), price * 1.001), style='.', color=colors[side][0], interactive=True)

and i guess what you might want is my function where my chart gets refreshed, aka the moment where i see _it_ disappear but not the normal static line:
[Here](https://prnt.sc/PVoCC6JzhT12)

**Comment by highfestiva** [comment, marked as answer]
In commit 3c0f3c63ac46654299ec3e5a40eaa92d4a649990, I added removal of primitives added previously. This means your lines will be destroyed properly. If you don't want them destroyed, you have a few options:
1. Don't call reset.
2. Remove the line prior to reset, put it back after.
3. Simply add the line again after reset.

I recommend you upgrade to 3c0f3c63ac46654299ec3e5a40eaa92d4a649990 unless you already have done so.

**Reply by seyeint** [reply]
i have to be honest, that part of the code where I reset is there just because this started with the help of an example that i found online.
what's the point of resetting in your view? does not reseting mean i will stack memory when plotting new information, even if i do see the new information?

i am using the update that is currently available when you search the package in the package manager, i'm not really pulling the project on every commit at the moment but i'm saving notes the rare couple of instances that i might have to update the name of a function if a new update on the package comes out.

**Reply by seyeint** [reply]
oh i see now... the reason i do need reset is that i have a combobox choosing my timeframe and if i change the timeframe, it does that update function and plots the plot above it, stacking candles with the previous timeframe... ok i will proceed with options 2 or 3. thanks

**Comment by seyeint** [comment]
btw why does finolyline does not have the attribute 'points' as the  finline? i'm trying to retrieve information from that previous line to help create it again as you proposed but i'm not being able to get its information

**Reply by highfestiva** [reply]
You need to clarify this question.
